src/steps/meta_commands/meta_command.ts

```diff
@@ -0,0 +1,31 @@
+export abstract class MetaCommand {
+    wrapperPath: string;
+    constructor(
+        public command: string | MetaCommand,
+        public args: string,
+        public timeout: number = Infinity,
+    ) {
+        this.wrapperPath = '';
+    }
+
+    serialize(): string {
+        if (typeof this.command === 'string') {
+            return `${this.wrapperPath} ${this.args} -- ${this.command}`;
+        }
+
+        return `${this.wrapperPath} ${
+            this.args
+        } -- ${this.command.serialize()}`;
+    }
+}
+
+export class RetryCommand extends MetaCommand {
+    constructor(
+        public command: string | MetaCommand,
+        public retries: number = 1,
+        public statusCode: number = 404,
```

Comment:
this seems to assume that there is some network operation involved? What about a command that needs to be retried out of the context of any network operarion?